mono/metadata/w32handle.c

```diff
@@ -49,6 +49,43 @@ static MonoW32HandleOps *handle_ops [MONO_W32HANDLE_COUNT];
 #define SLOT_INDEX(x)	(x / HANDLE_PER_SLOT)
 #define SLOT_OFFSET(x)	(x % HANDLE_PER_SLOT)
 
+static struct {
+	gint64 handle_count;
+	gint64 total_handles_created;
+	gint64 maximum_handle_number;
+	gboolean rescanned;
+} handle_metrics = { 0 } ;
+
+static struct {
+	gboolean in_use;
+} handles_debug_table [SLOT_MAX * HANDLE_PER_SLOT] = {{0}};
```

Comment:
This adds 32MB to the size of the executable or the heap. Either have the same mechanism as `private_handles` with a double array (`[][]`), or use a hash table (it would need locking).